File: dto-mapping-benchmark/src/main/java/com/example/dtomappingbenchmark/v4/service/UserService.java
package com.example.dtomappingbenchmark.v4.service;

import com.example.dtomappingbenchmark.v4.dto.UserDto;
import com.example.dtomappingbenchmark.v4.entity.User;
import com.example.dtomappingbenchmark.v4.mapper.UserMapper;
import com.example.dtomappingbenchmark.v4.repository.UserRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
public class UserService {
    private final UserRepository userRepository;

    public UserDto create(UserDto dto) {
        User user = UserMapper.toEntity(dto);
        return UserMapper.toDto(userRepository.save(user));
    }

    @Transactional
    public UserDto get(Long id) {
        return UserMapper.toDto(userRepository.findById(id).orElseThrow());
    }
}

File: object-mapping-benchmark-v5/src/main/kotlin/yu/selab/cj/objectmappingbenchmarkv5/user/service/UserService.kt
package yu.selab.cj.objectmappingbenchmarkv5.user.service


import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional
import yu.selab.cj.objectmappingbenchmarkv5.mapper.UserMapper
import yu.selab.cj.objectmappingbenchmarkv5.user.command.CreateUserDto
import yu.selab.cj.objectmappingbenchmarkv5.user.command.ReadUserDto
import yu.selab.cj.objectmappingbenchmarkv5.user.repository.UserRepository


@Service
class UserService(
    private val userRepository : UserRepository,
    private val userMapper : UserMapper,
) {

    @Transactional(readOnly = true)
    fun getUserList() = userRepository.findAll().map{
        user -> userMapper.toDto(user)
    }

    @Transactional(readOnly = true)
    fun getUser(id : Long) : ReadUserDto {

        val user = userRepository.findById(id).orElseThrow {
            IllegalArgumentException("유저 없음")
        }

        return userMapper.toDto(user)
    }

    @Transactional
    fun createUser(dto : CreateUserDto){
        val user = userMapper.toEntity(dto)
        userRepository.save(user)
    }

    @Transactional
    fun deleteUser(id : Long)
        = userRepository.deleteById(id)

}